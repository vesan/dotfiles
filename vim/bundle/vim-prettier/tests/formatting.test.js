const fs = require('fs');
const path = require('path');
const HeadlessRemoteClient = require('vim-driver/dist/HeadlessRemoteClient');
const Server = require('vim-driver/dist/Server');

const HOST = '127.0.0.1';
const PORT = 1337;
const FIXTURES_DIR = `${__dirname}/fixtures`;

let server;
let remote;

jest.setTimeout(15000);

const getBufferContents = async remote =>
  (await remote.call('getline', [1, '$'])).join('\n');

const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
const waitUntil = (condition, timeout = 2000) => {
  return new Promise(resolve => {
    let isTimedOut = false;
    let timeoutId = null;

    const check = () => {
      const promise = condition();
      promise.then(result => {
        if (!isTimedOut && result === true) {
          clearTimeout(timeoutId);
          resolve();
        } else if (!isTimedOut) {
          check();
        }
      });
    };

    timeoutId = setTimeout(() => {
      isTimedOut = true;
      resolve();
    }, timeout);
  });
};

const assertFormatting = (file) => {
  const filename = path.basename(file);
  test(`Prettier formats ${filename} file with :Prettier command`, async () => {
    await remote.edit(`${FIXTURES_DIR}/${file}`);

    const lines = await getBufferContents(remote);

    // run sync formatting
    await remote.execute('Prettier');

    const updatedLines = await getBufferContents(remote);

    // we now check that we have indeed formatted the code
    expect(updatedLines).not.toBe(lines);

    // check snapshot
    expect(updatedLines).toMatchSnapshot();
  });

  test(`Prettier formats ${filename} file with :PrettierAsync command`, async () => {
    await remote.edit(`${FIXTURES_DIR}/${file}`);

    const lines = await getBufferContents(remote);

    // run async formatting
    await remote.execute('PrettierAsync');

    const unmodifiedLines = await getBufferContents(remote);

    // async should not happen immediatly so content should still remain the same
    expect(unmodifiedLines).toBe(lines);

    // we now will wait until prettier has finally updated the content async
    await waitUntil(async () => (await getBufferContents(remote)) !== lines);

    const updatedLines = await getBufferContents(remote);

    // we now check that we have indeed formatted the code
    expect(lines).not.toBe(updatedLines);

    // check snapshot
    expect(updatedLines).toMatchSnapshot();
  });
};

beforeAll(async () => {
  server = new Server();
  await server.listen(HOST, PORT);
});

afterAll(async () => {
  await server.close();
});

// should ensure that we cache original fixture contents and
// restore it on the afterEach
beforeEach(async () => {
  remote = new HeadlessRemoteClient({host: HOST, port: PORT});
  await remote.connect(server);
});

afterEach(async () => {
  if (remote.isConnected()) {
    try {
      const filename = await remote.call('expand', ['%:p']);

      if (filename) {
        // restore the file
        await remote.execute('earlier 1d | noautocmd | write');
      }
    } catch (e) {
    } finally {
      await remote.close();
    }
  }
});

//test('PrettierVersion returns pluggin version', async () => {
//  const result = await remote.execute('PrettierVersion');
//  expect(result).toMatchSnapshot();
//});

// run formatting tests in all fixtures
fs.readdirSync(FIXTURES_DIR).forEach(file => assertFormatting(file));
