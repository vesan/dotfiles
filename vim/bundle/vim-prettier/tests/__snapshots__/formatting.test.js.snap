// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prettier formats foo.css file with :Prettier command 1`] = `
".redClass {
    background: \\"red\\";
}
#redId {
    background: \\"red\\";
}"
`;

exports[`Prettier formats foo.css file with :PrettierAsync command 1`] = `
".redClass {
    background: \\"red\\";
}
#redId {
    background: \\"red\\";
}"
`;

exports[`Prettier formats foo.graphql file with :Prettier command 1`] = `
"{
    baz {
        blah
        ...name
    }
    bean {
        foo
        blah
        bleh
    }
    bar {
        ...name
    }
}"
`;

exports[`Prettier formats foo.graphql file with :PrettierAsync command 1`] = `
"{
    baz {
        blah
        ...name
    }
    bean {
        foo
        blah
        bleh
    }
    bar {
        ...name
    }
}"
`;

exports[`Prettier formats foo.html file with :Prettier command 1`] = `
"<html>
    <head>
        <title>foo</title>
    </head>
    <body>
        <h1>foo</h1>
    </body>
</html>"
`;

exports[`Prettier formats foo.html file with :PrettierAsync command 1`] = `
"<html>
    <head>
        <title>foo</title>
    </head>
    <body>
        <h1>foo</h1>
    </body>
</html>"
`;

exports[`Prettier formats foo.js file with :Prettier command 1`] = `
"const a = () => {
    return 2;
};"
`;

exports[`Prettier formats foo.js file with :PrettierAsync command 1`] = `
"const a = () => {
    return 2;
};"
`;

exports[`Prettier formats foo.json file with :Prettier command 1`] = `
"{
    \\"name\\": \\"foo\\",
    \\"bar\\": \\"baz\\"
}"
`;

exports[`Prettier formats foo.json file with :PrettierAsync command 1`] = `
"{
    \\"name\\": \\"foo\\",
    \\"bar\\": \\"baz\\"
}"
`;

exports[`Prettier formats foo.less file with :Prettier command 1`] = `
".redClass {
    background: \\"red\\";
}
#redId {
    background: \\"red\\";
}"
`;

exports[`Prettier formats foo.less file with :PrettierAsync command 1`] = `
".redClass {
    background: \\"red\\";
}
#redId {
    background: \\"red\\";
}"
`;

exports[`Prettier formats foo.lua file with :Prettier command 1`] = `
"function deepcopy(orig)
	local orig_type = type(orig)
	local copy

	return copy
end"
`;

exports[`Prettier formats foo.lua file with :PrettierAsync command 1`] = `
"function deepcopy(orig)
	local orig_type = type(orig)
	local copy

	return copy
end"
`;

exports[`Prettier formats foo.md file with :Prettier command 1`] = `
"## foo

Lorem ipsum dolor amet

### bar

-   a
-   b
-   c"
`;

exports[`Prettier formats foo.md file with :PrettierAsync command 1`] = `
"## foo

Lorem ipsum dolor amet

### bar

-   a
-   b
-   c"
`;

exports[`Prettier formats foo.php file with :Prettier command 1`] = `
"<?php
$a = 'a';

$b = $a . $a . ' asda';"
`;

exports[`Prettier formats foo.php file with :PrettierAsync command 1`] = `
"<?php
$a = 'a';

$b = $a . $a . ' asda';"
`;

exports[`Prettier formats foo.rb file with :Prettier command 1`] = `
"[1, 2, 3].each do |i|
    p i
end"
`;

exports[`Prettier formats foo.rb file with :PrettierAsync command 1`] = `
"[1, 2, 3].each do |i|
    p i
end"
`;

exports[`Prettier formats foo.scss file with :Prettier command 1`] = `
".redClass {
    background: \\"red\\";
}
#redId {
    background: \\"red\\";
}"
`;

exports[`Prettier formats foo.scss file with :PrettierAsync command 1`] = `
".redClass {
    background: \\"red\\";
}
#redId {
    background: \\"red\\";
}"
`;

exports[`Prettier formats foo.ts file with :Prettier command 1`] = `
"const a: () => number = () => {
    return 2;
};"
`;

exports[`Prettier formats foo.ts file with :PrettierAsync command 1`] = `
"const a: () => number = () => {
    return 2;
};"
`;

exports[`Prettier formats foo.vue file with :Prettier command 1`] = `
"<template>
    <div class=\\"example\\">{{ msg }}</div>
</template>

<script>
export default {
    data() {
        return { msg: \\"Hello world!\\" };
    }
};
</script>

<style>
.example {
    color: red;
}
</style>

<custom1>
  This could be e.g. documentation for the component.
</custom1>"
`;

exports[`Prettier formats foo.vue file with :PrettierAsync command 1`] = `
"<template>
    <div class=\\"example\\">{{ msg }}</div>
</template>

<script>
export default {
    data() {
        return { msg: \\"Hello world!\\" };
    }
};
</script>

<style>
.example {
    color: red;
}
</style>

<custom1>
  This could be e.g. documentation for the component.
</custom1>"
`;

exports[`Prettier formats foo.xml file with :Prettier command 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<!DOCTYPE module PUBLIC \\"-//Puppy Crawl//DTD Check Configuration 1.3//EN\\"
    \\"https://www.puppycrawl.com/dtds/configuration_1_3.dtd\\">
<?xml-model href=\\"project.rnc\\" type=\\"application/relax-ng-compact-syntax\\"?>
<!-- foo -->
<svg
    xmlns=\\"http://www.w3.org/2000/svg\\"
    xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"
    width=\\"200\\"
    height=\\"100\\"
    viewBox=\\"0 0 200 100\\"
>
  <title>foo</title>
  <desc _attr=\\"attr\\">
    bar
  </desc>
  <style />
</svg>
<!-- bar -->"
`;

exports[`Prettier formats foo.xml file with :PrettierAsync command 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<!DOCTYPE module PUBLIC \\"-//Puppy Crawl//DTD Check Configuration 1.3//EN\\"
    \\"https://www.puppycrawl.com/dtds/configuration_1_3.dtd\\">
<?xml-model href=\\"project.rnc\\" type=\\"application/relax-ng-compact-syntax\\"?>
<!-- foo -->
<svg
    xmlns=\\"http://www.w3.org/2000/svg\\"
    xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"
    width=\\"200\\"
    height=\\"100\\"
    viewBox=\\"0 0 200 100\\"
>
  <title>foo</title>
  <desc _attr=\\"attr\\">
    bar
  </desc>
  <style />
</svg>
<!-- bar -->"
`;

exports[`Prettier formats foo.yaml file with :Prettier command 1`] = `
"foo:
    bar:
        baz:
            bleh: ./foo
            foobar: foobar
            args:
                buildno: 1"
`;

exports[`Prettier formats foo.yaml file with :PrettierAsync command 1`] = `
"foo:
    bar:
        baz:
            bleh: ./foo
            foobar: foobar
            args:
                buildno: 1"
`;
